java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -526,4 +625,199 @@ public void negativeCaseWithFlagNameAsVariable() throws IOException {
       assert_().fail("Incorrect parameters passed to the checker");
     }
   }
+
+  @Test
+  public void noFlagCase() throws IOException {
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "config/properties.json");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+          BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+      bcr = addHelperClasses(bcr);
+      bcr.addInputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  // BUG: Diagnostic contains: Cleans stale XP flags",
+              "  if (experimentation.isToggleEnabled()) { int a = 1; }",
+              "     else { int b = 2;}",
+              "  if (experimentation.isUnrelatedToggleEnabled()) { int c = 1; }",
+              "     else { int d = 2;}",
+              "  if (experimentation.isToggleDisabled()) { return \"X\"; }",
+              "     else { return \"Y\";}",
+              " }",
+              "}")
+          .addOutputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "     int a = 1;",
+              "  if (experimentation.isUnrelatedToggleEnabled()) { int c = 1; }",
+              "     else { int d = 2;}",
+              "     return \"Y\";",
+              " }",
+              "}");
+
+      bcr.doTest();
+    } catch (ParseException pe) {
+      pe.printStackTrace();
+      assert_().fail("Incorrect parameters passed to the checker");
+    }
+  }
+
+  @Test
+  public void noFlagCaseWithFlagSpecified() throws IOException {
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "config/properties.json");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+          BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+      bcr = addHelperClasses(bcr);
+      bcr.addInputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  // BUG: Diagnostic contains: Cleans stale XP flags",
```

Comment:
Should this produce an error here if it's not going to refactor anything?

Reply:
Looking at the code, seems like it's supposed to 🤔 
But locally I don't get any error, irrespective of that line (this is true for all of my tests btw)